onnxruntime/core/providers/dml/dml_provider_factory.cc

```diff
@@ -196,6 +233,18 @@
   return nullptr;
 }
 
+bool isHardwareAdapter(ComPtr<IDXCoreAdapter> adapter) {
+    bool isHardware{ false };
+    THROW_IF_FAILED(adapter->GetProperty(
+        DXCoreAdapterProperty::IsHardware,
+        &isHardware));
+    return isHardware;
+}
+
+bool supportsGraphics(ComPtr<IDXCoreAdapter> adapter) {
```

Comment:
>ComPtr<IDXCoreAdapter> adapter [](http://example.com/codeflow?start=22&length=30)

passing by value here will cause addref/release.

instead pass in IDXCoreAdapter* which will avoid pointer tracking